doc/basics/tasks_tutorial.rst

```diff
@@ -0,0 +1,168 @@
+   :lines: 11-16
+
+As an added bonus, this is also faster than the previous method.
+This new example will finish when the ``execute_me_too(4)`` call completes, instead of after waiting a full second.
+
+This concludes the basics of ``asyncio`` tasks.
+Now, we'll show you how to make your life easier by using IPv8's ``TaskManager`` class.
+
+The TaskManager
+---------------
+
+Managing ``Future`` instances and ``coroutine`` instances is complex and error-prone and, to help you with managing these, IPv8 has the ``TaskManager`` class.
+The ``TaskManager`` takes care of managing all of your calls that have not completed yet and allows you to cancel them on demand.
+You can even find this class in a separate PyPi repository: https://pypi.org/project/ipv8-taskmanager/
+
+Adding tasks
+^^^^^^^^^^^^
+
+To add tasks to your ``TaskManager`` you can call ``register_task()``.
+You register a task with a name, which you can later use to inspect the state of a task or cancel it.
+For example:
+
+.. literalinclude:: tasks_tutorial_3.py
+
+This example prints ``[2]``.
+If you had not called ``wait_for_tasks()`` before ``shutdown_task_manager()`` in this example, all of your registered tasks would have been canceled, printing ``[]``.
+
+In some cases you may also not be interested in canceling a particular task.
+For this use case the ``register_anonymous_task()`` call exists, for example:
+
+.. literalinclude:: tasks_tutorial_4.py
+   :lines: 16-19
+
+Note that this example takes just over half a second to execute, all 20 calls to ``execute_me_too()`` are scheduled at (almost) the same time!
+
+Periodic and delayed tasks
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next to simply adding tasks, the ``TaskManager`` also allows you to invoke calls after a delay or periodically.
+The following example will add the value ``1`` to the ``COMPLETED`` list periodically and inject a single value of ``2`` after half a second:
+
+.. literalinclude:: tasks_tutorial_5.py
+   :lines: 8-26
+
+This example prints ``[1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]``.
+
+Note that you can also create a task with both an interval and a delay.
+
+Community Integration
+^^^^^^^^^^^^^^^^^^^^^
+
+For your convenience, every ``Community`` is also a ``TaskManager``.
+However, try to avoid spawning any tasks in the ``__init__`` of your ``Community``.
+Specifically, you could end up scheduling a task in between your ``Community`` and other ``Community`` initialization
+Working with a half-initialized IPv8 may lead to problems.
+This practice also makes your ``Community`` more difficult to test.
+
+If you do end up in a situation where you absolutely must schedule a task from the ``__init__()``, you can use the ``TaskManager``'s ``cancel_all_pending_tasks()`` method to cancel all registered tasks.
+This is an example of how to deal with this in your unit tests:
+
+.. literalinclude:: tasks_tutorial_6.py
+   :lines: 18-26
+
+Network IO and the DiscoveryStrategy
```

Comment:
This part feels a bit detached from the overall story which is mostly about the task manager. Maybe you could move it to a separate tutorial file?